src/main/java/org/wildfly/security/auth/login/ServerAuthenticationContext.java

```diff
@@ -233,6 +288,65 @@ public void fail() throws IllegalStateException {
     }
 
     /**
+     * Attempt to authorize an authentication attempt.  If the authorization is successful (meaning, the authenticated
+     * user is the same as the authorize user, or otherwise possesses a sufficient {@link RunAsPrincipalPermission}),
+     * {@code true} is returned and the context is placed in the "successful" state as if {@link #succeed()} had been
+     * called.  If the authorization fails, {@code false} is returned and the context is placed in the "failed" state
+     * as if {@link #fail()} had been called.
+     *
+     * @param name the authorization name
+     * @return {@code true} if the authorization succeeded, {@code false} otherwise
+     * @throws IllegalArgumentException if the name is syntactically invalid
+     * @throws RealmUnavailableException if the realm is not available
+     * @throws IllegalStateException if the authentication name was not set or authentication was already complete
+     */
+    public boolean authorize(String name) throws IllegalArgumentException, RealmUnavailableException, IllegalStateException {
+        State oldState = stateRef.get();
+        if (oldState.isDone()) {
+            throw ElytronMessages.log.alreadyComplete();
+        }
+        if (! oldState.isStarted()) {
+            throw ElytronMessages.log.noAuthenticationInProgress();
+        }
+        Assert.checkNotNullParam("name", name);
+        name = domain.getPreRealmRewriter().rewriteName(name);
+        String realmName = domain.getRealmMapper().getRealmMapping(name);
+        if (realmName == null) {
+            realmName = domain.getDefaultRealmName();
+        }
+        Principal principal = new NamePrincipal(name);
+        if (oldState.getAuthenticationPrincipal().equals(principal)) {
```

Comment:
It is similar but not identical.  We need the realmInfo and other pieces for later, and we don't to have a case where we look up that stuff once and then look it up again (possibly getting a different result).